src/js/CliAutoComplete.js

```diff
@@ -0,0 +1,482 @@
+'use strict';
+
+/**
+ * Encapsulates the AutoComplete logic
+ *
+ * Uses: https://github.com/yuku/jquery-textcomplete
+ * Check out the docs at https://github.com/yuku/jquery-textcomplete/tree/v1/doc
+ */
+const CliAutoComplete = {
```

Comment:
I am normally trying to keep the configurator coding style as close as possible to C coding style. This seems to make sense since the 'main' project is the firmware which is in C, and we want to make it as easy as possible for contributors to the firmware to make the changes to configurator that are required to match their changes in the firmware themselves.

This approach would call for constants to be named in MACRO_CASE.

Reply:
Sure,
Actually I've never used the `const` keyword in `javascript` before. I just saw that it is in use in the existing code and decided to go along.
About the MACRO_CASE - I'm also used to this, but I thought that in `javascript` `const` is used more in the sense of `final` as in `Java`.

What would you recommend? Simple to do it as `var`s, unless it's a real constant primitive (which will be uppercased)?